contrib/generate_precompile.jl

```diff
@@ -0,0 +1,112 @@
+Base.__init__()
+
+# Prevent this from being put into the Main namespace
+let
+M = Module()
+@eval M begin
+julia_cmd() = (julia = joinpath(Sys.BINDIR, Base.julia_exename()); `$julia`)
+
+function generate_precompilable_package(path)
+    pkgname = "Package"
+    mkpath(joinpath(path, pkgname, "src"))
+    write(joinpath(path, pkgname, "src", "Package.jl"),
+          """
+          __precompile__()
+          module Package
+          end
+          """)
+end
+
+function generate_precompile_statements()
+    t = time()
+    println("Generating precompile statements...")
+    println("──────────────────────────────────────")
+    tmpd = mktempdir()
+
+	# Reset code loading vars
+	Base.ACTIVE_PROJECT[] = nothing
+	Base.HOME_PROJECT[] = nothing
+	empty!(LOAD_PATH)
+	empty!(DEPOT_PATH)
+	push!(LOAD_PATH, "@stdlib")
+
+    push!(DEPOT_PATH, tmpd)
+    push!(LOAD_PATH, tmpd)
+    generate_precompilable_package(tmpd)
+    @eval using Package
+    pop!(LOAD_PATH)
+    pop!(DEPOT_PATH)
+
+    if isempty(ARGS)
+        sysimg = joinpath(dirname(Sys.BINDIR), "lib", "julia", "sys.ji")
+    else
+        sysimg = ARGS[1]
+    end
+
+    tmp = tempname()
+    touch(tmp)
+    have_repl =  haskey(Base.loaded_modules,
+                        Base.PkgId(Base.UUID("3fa0cd96-eef1-5676-8a61-b3b8758bbffb"), "REPL"))
+    if have_repl
+        # Have a REPL, run the repl replayer and an interactive session that we immidiately kill
+        setup = """
+        include($(repr(joinpath(@__DIR__, "precompile_replay.jl"))))
+        @async while true
+            sleep(0.01)
+            if isdefined(Base, :active_repl)
+                exit(0)
+            end
+        end
+        """
+        # Do not redirect stdin unless it is to a tty, because that changes code paths
+        run(pipeline(`$(julia_cmd()) --sysimage $sysimg --trace-compile=yes -O0
```

Comment:
Is there something faster than -O0 here?

Reply:
The other option is --compile=min, which uses the interpreter for almost everything.

Reply:
Tried it and it never finished.